deploy/helm/kubecf/values.yaml

```diff
@@ -328,3 +330,124 @@ k8s-host-url: ""
 k8s-service-token: ""
 k8s-service-username: ""
 k8s-node-ca: ""
+
+eirini:
+  opi:
+    use_registry_ingress: false
+    ingress_endpoint: ~
+    kube:
+      external_ips: []
+    services:
+      loadbalanced: false
+
+    deny_app_ingress: false
+
+    cc_api:
+      serviceName: "kubecf-api"
```

Comment:
The `kubecf` prefix here and below only works incidentally due to the Helm installation name being `kubecf`. I.e. if someone deploys KubeCF with `helm install kubecf-experiment ...`, it blows up!

Reply:
I don't have a solution for this... we need to think.

Reply:
This is indeed true. The default is set to `kubecf`, because in general the it is deployed with this release name. Users could overwrite the configuration in case the release name deviates. Which is not nice.
The problem might be solvable by not setting the secret names based on the release-name, but have a common prefix for the charts (which could be kubecf). We were wondering why the kubecf helm charts are designed like this in first place? Can this be changed?

Reply:
@mook-as pointed out to me that `.Release.Name` is still available on the sub-chart. If we change the sub-charts to call the `tpl` function on the values, we can do magic like:

```yaml
# -----------------------------------
# Sub-chart bar
apiVersion: v1
kind: Secret
metadata:
  name: {{ tpl .Values.secretName . }}

# -----------------------------------
# values.yaml of foo
bar:
  secretName: '{{ printf "%s.var-tls" .Release.Name }}'
```

`helm template foo-experiment .` renders to:

```yaml
---
# Source: foo/charts/bar/templates/secret.yaml
apiVersion: v1
kind: Secret
metadata:
  name: foo-experiment.var-tls
```

This would require the Eirini and bits-service Helm charts to allow this trick.

Reply:
Hey @f0rmiga! I think from our point of view we would like to change as little as possible in the Eirini helm chart and make it as generic as possible. Our current goal is to detach from all cf-on-top-of-k8s deployment methods so we (1) can concentrate on providing a concise Eirini helm chart and (2) can be easily consumed by whoever. Having said that, I am a bit reluctant to make such implementation-specific changes to our helm chart because I don’t know how this will play out with cf-for-k8s (which we also support).

Have you considered removing the `.Release.name` prefix from all kubecf components? This seems like the proper approach to me. I’m not sure what value having this prefix has for you. It might also cause similar issues in the future when you consume other sub-charts.

Reply:
Hey! Adding `tpl` wouldn't cause any side-effects for other users. But I understand that such a change is not normal and that we most likely will have problems with other charts.

@viovanov Could we revisit the prefixes added by cf-operator to k8s resources?

Reply:
@f0rmiga we could - those are "bosh deployment names" though, not helm release names.

Reply:
https://www.pivotaltracker.com/story/show/171854659

I wouldn't block this on this issue @f0rmiga

Reply:
@viovanov, the BOSH deployment name and the Helm release name happen to match for us. Thanks for creating the story. I'll add the blocked label.